bin/p2-exec/main.go

```diff
@@ -37,6 +38,8 @@ func main() {
 	kingpin.Version(version.VERSION)
 	kingpin.Parse()
 
+	syscall.Umask(*umask)
```

Comment:
I'd feel better about this being optional. If the flag is set, change the umask. Otherwise, don't do anything. That's how all the other flags in `p2-exec` work.